File: src/main/kotlin/org/bytecamp/program_repair/backend/utils/FileWriter.java
package org.bytecamp.program_repair.backend.utils;

import java.io.File;
import java.io.IOException;

public class FileWriter implements IWriter {
    final private java.io.Writer file;
    public FileWriter(File file) throws IOException {
        this.file = new java.io.FileWriter(file);
    }
    @Override
    public void write(String chunk) {
        try {
            file.write(chunk);
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}

File: src/main/kotlin/org/bytecamp/program_repair/backend/utils/FileWriter.kt
package org.bytecamp.program_repair.backend.utils

import java.io.File
import java.io.FileWriter
import java.io.IOException
import java.io.Writer

class FileWriter(file: File) : IWriter {
    private val file: Writer
    override fun write(chunk: String) {
        try {
            file.write(chunk)
        } catch (e: IOException) {
            e.printStackTrace()
        }
    }

    init {
        this.file = FileWriter(file)
    }
}